stream_framer.go

```diff
@@ -71,33 +71,32 @@ func (f *streamFramer) PopCryptoStreamFrame(maxLen protocol.ByteCount) *wire.Str
 	return frame
 }
 
-func (f *streamFramer) maybePopFramesForRetransmission(maxLen protocol.ByteCount) (res []*wire.StreamFrame, currentLen protocol.ByteCount) {
+func (f *streamFramer) maybePopFramesForRetransmission(maxTotalLen protocol.ByteCount) (res []*wire.StreamFrame, currentLen protocol.ByteCount) {
 	for len(f.retransmissionQueue) > 0 {
 		frame := f.retransmissionQueue[0]
 		frame.DataLenPresent = true
 
 		frameHeaderLen, _ := frame.MinLength(f.version) // can never error
-		if currentLen+frameHeaderLen >= maxLen {
+		maxLen := maxTotalLen - currentLen
+		if maxLen < protocol.MinStreamFrameSize {
```

Comment:
At least for the retransmissions, we should probably change this to

`if frameHeaderLen + frame.DataLen() > maxLen && maxLen < protocol.MinStreamFrameSize`